src/main/java/com/pusher/client/channel/impl/PrivateEncryptedChannelImpl.java

```diff
@@ -0,0 +1,148 @@
+package com.pusher.client.channel.impl;
+
+import com.google.gson.Gson;
+import com.pusher.client.AuthorizationFailureException;
+import com.pusher.client.Authorizer;
+import com.pusher.client.channel.PrivateEncryptedChannel;
+import com.pusher.client.channel.PrivateEncryptedChannelEventListener;
+import com.pusher.client.channel.SubscriptionEventListener;
+import com.pusher.client.connection.impl.InternalConnection;
+import com.pusher.client.crypto.nacl.SecretBoxOpener;
+import com.pusher.client.util.Base64;
+import com.pusher.client.util.Factory;
+
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class PrivateEncryptedChannelImpl extends ChannelImpl implements PrivateEncryptedChannel {
+
+    private static final Gson GSON = new Gson();
+    private final InternalConnection connection;
+    private final Authorizer authorizer;
+    private SecretBoxOpener secretBoxOpener;
+
+    private class PrivateEncryptedChannelData {
+        byte[] auth;
+        final String channelData;
```

Comment:
What's in the `channelData`?

Reply:
Good question - I [wrote a comment](https://github.com/pusher/pusher-websocket-java/pull/234#discussion_r398505881) to explain my reasoning for why I created it.

Reply:
I believe Mike's question here was about `String channelData` not its enclosing class (which btw has a clashing name but I will provide more in-depth feedback on it in a septate, more relevant and hopefully more illustrative place in the diff). At least I had the same question and I've checked the docs of what `String channelData` is. It's only [present in presence channels](https://pusher.com/docs/channels/library_auth_reference/auth-signatures#presence-channels) (pun semi-intended;), and [not present in encrypted channels](https://pusher.com/docs/channels/library_auth_reference/auth-signatures#encrypted-channels). That should be removed to simplify the code and to not create confusion.

Reply:
Hmm. It was used in the subscribeTo method, so i thought it was needed. But it was always empty when I inspected. I have removed this and refactored. We don't need the `PrivateEncryptedChannelData` data class anymore then, and the `checkAuthentication` method returns the auth key.